### ISSUE TITLE
2.0-preview does not fire iron-form-invalid event

### ISSUE THREAD

**Issue by jehna** [PolymerElements/iron-form#176]
### Description

The `2.0-preview` branch does not fire `iron-form-invalid` event when the form's validation fails.

### Expected outcome

The form should fire `iron-form-invalid` event when the form is validated and the form's validation fails.

### Actual outcome

The `iron-form-invalid` event is not fired.

**Comment by jehna** [comment]
Quick fix for the bug:

```
this.$.youFormId._validate = this.$.youFormId.validate
this.$.youFormId.validate = () => this.$.youFormId._validate() || this.$.youFormId.dispatchEvent(new Event('iron-form-invalid'))
```

**Comment by MWalid** [comment]
Even `iron-form-presubmit` is not fired...

**Comment by notwaldorf** [comment]
Fixed in a8e7993
